src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/C14NAncestralNamespaceContextManager.cs

```diff
@@ -89,9 +89,9 @@ internal override void TrackNamespaceNode(XmlAttribute attr, SortedList nsListTo
             nsLocallyDeclared.Add(Utils.GetNamespacePrefix(attr), attr);
         }
 
-        internal override void TrackXmlNamespaceNode(XmlAttribute attr, SortedList nsListToRender, SortedList attrListToRender, Hashtable nsLocallyDeclared)
+        internal override void TrackXmlNamespaceNode(XmlAttribute? attr, SortedList nsListToRender, SortedList attrListToRender, Hashtable nsLocallyDeclared)
         {
-            nsLocallyDeclared.Add(Utils.GetNamespacePrefix(attr), attr);
+            nsLocallyDeclared.Add(Utils.GetNamespacePrefix(attr!), attr);
```

Comment:
Here `attr` shouldn't be nullable too

Reply:
The override forces us to make the parameter nullable. But `GetNamespacePrefix` doesn't expect a null. We can't change the signature of the base class, nor the signature of `GetNamespacePrefix`. I can't see an elegant way around this without changing anything outside the remit of these PRs.

Reply:
> We can't change the signature of the base class
Why so? `AncestralNamespaceContextManager` is within this project, so you can change it's annotations

Reply:
Very valid point. I just double checked and the base method is non null. Well spotted! Thanks.